torch_xla/launch.py

```diff
@@ -0,0 +1,49 @@
+import warnings
+
+import torch.distributed as dist
+import torch_xla
+import torch_xla.core.xla_env_vars as xenv
+import torch_xla.distributed.xla_multiprocessing as xmp
+import torch_xla.utils.utils as xu
+import torch_xla.runtime as xr
+
+
+def spawn_wrapper(index, fn, *args, **kwargs):
+  dist.init_process_group('xla', init_method='xla://')
+  fn(index, *args, **kwargs)
+
+
+def launch(
+    fn,
+    *args,
+    **kwargs,
+):
+  """ Entry to launch multiprocess.
+
+  Raises:
+    NotImplementedError: SPMD is not supported yet.
+  """
+  if xr.is_spmd():
+    # TODO(piz): SPMD is specified differently from mp. Skip for now.
+    raise NotImplementedError(
+        'launch function does not support SPMD at this time')
+
+  nprocs = kwargs.pop('nprocs', None)
```

Comment:
Put only the options that are implemented here directly as keyword arguments. Namely, `spawn_method` and `nprocs`

The only valid states of `nprocs` are 1 and `num_devices`. You can make the illegal states impossible by replacing it with a bool (e.g. `debug_single_process`) that determines whether we pass `1` to `xmp.spawn` or not.

`join` and `daemon` are not implemented in `xmp.spawn` and were just there to mirror torch's `spawn`.

Reply:
Regarding `The only valid states of nprocs are 1 and num_devices` , is this also true for GPU?